### ISSUE TITLE
🐛 BUG: head contents placed in body when page has conditional head

### ISSUE THREAD

**Issue by IanVS** [withastro/astro#2046]
### What version of `astro` are you using?

0.21.4

### What package manager are you using?

npm

### What operating system are you using?

mac

### Describe the Bug

I found that my utf-8 encoding was not being applied correctly, and I traced it down to my head contents being placed inside the `body` instead of in the head.  It occurs due to my page looking like this:

```
{redirect 
  ? <head><meta http-equiv="refresh" content={`0; url=${redirect}`}></head>
  : <Layout>
      It’s a trap!
    </Layout>
}
```

If I change the redirect to a normal layout or something, it works.  So I suppose I just need to make another component to do the redirect.  But I wanted to at least let y'all know this is happening.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-v7dk72?file=src/pages/index.astro

**Comment by FredKSchott** [comment]
Looks like you got a fix in that commit, but +1 to a larger fix for us. If we can't fix this as a small bug, we're also getting ready to redo our larger head handling soon. RFC here: https://github.com/withastro/rfcs/pull/26

**Comment by tony-sull** [comment]
An [RFC](https://github.com/withastro/rfcs/blob/main/proposals/0007-finalize-head-body.md) was recently accepted to finalize `<head>` and `<body>` behavior :+1:

**Comment by natemoo-re** [comment]
This is on track to be fixed in https://github.com/withastro/compiler/pull/267 and the subsequent `0.24.x` release of `astro`.

**Comment by natemoo-re** [comment]
This should be fixed in the newly released `astro@0.24.0`! 

Please give feel free to reopen this issue if you're still having trouble.
